src/types/boolobject.rs

```diff
@@ -55,7 +55,7 @@ impl PyBool {
 /// syntax these methods are separated into a trait, because stable Rust does not yet support
 /// `arbitrary_self_types`.
 #[doc(alias = "PyBool")]
-pub trait PyBoolMethods<'py> {
+pub trait PyBoolMethods<'py>: crate::sealed::Sealed {
```

Comment:
Does documentation render the bound as `Sealed` or `crate::sealed::Sealed`? That might affect whether we `use` the trait first or go for the full path like here (if you think it matters).

Reply:
![image](https://github.com/PyO3/pyo3/assets/10486322/22cbd411-d33e-4ff5-84e4-601df60ef38d)
It seems it does not make a difference, this looks fine to me.